### ISSUE TITLE
String IDs cant be used in hooked init functions of layers

### ISSUE THREAD

**Issue by Jollycistaken** [geode-sdk/geode#59]
### Platform

Windows

### Geode Version

I think its 0.6.0

### Mods Installed

geode.DevTools and geode.Testing (my testing thing)

### Expected Behavior

when you run getID() on a node with a string id on it, it should return the string id

### Actual Behavior

when I run it, it just returns a empty string

### Steps to Reproduce

you can do
```c++
class $modify(MenuLayer) {
    bool init() {
	if (!MenuLayer::init()) return false;
        auto menuid = this->getID();
	return true;
    }
};
```
and you can print out the menuid by any means and it should be empty or at least for me it is empty

**Comment by altalk23** [comment]
Known bug, will hopefully be fixed in version 1.0
We basically need the new hook system to fix it (hook ordering)

Ig we can patch something until 1.0 but yeah

**Comment by Jollycistaken** [comment]
ah ok

**Comment by Jollycistaken** [comment]
did not mean to close it

**Comment by Jollycistaken** [comment]
just going to leave this up until it gets fixed

**Comment by altalk23** [comment]
Fixed with #116
